ui/app/utils/common-prefix.js

```diff
@@ -0,0 +1,24 @@
+import { get } from '@ember/object';
+
+export default function(arr, attribute = 'id') {
+  let content = arr || [];
+  // this assumes an already sorted array
+  // if the array is sorted, we want to compare the first and last
+  // item in the array - if they share a prefix, all of the items do
+  let firstString = get(content[0], attribute);
+  let lastString = get(content[arr.length - 1], attribute);
+
```

Comment:
I'm confused as to why you only check the last string in the array here, maybe I don't understand the use case? Ah ok hold on I just re-read that comment up above here (lines 5-8). Wonder if it would be nicer to just pass the two strings instead of an array?
```javascript
function commonPrefix(a = '', b = '') {
...
}
commonPrefix(arr[0].id, arr[arr.length - 1].id)
commonPrefix(arr[0].name, arr[arr.length - 1].name)
// maybe I might want to recur this at some point
function(item) {
  commonPrefix(prev.id, item.id)
}
```

You also remove your `get` dependency by doing that and reduces your function to just do one thing. Again, up to you

Reply:
For now I think having the whole array contains the overall logic a bit better - if we later decide to change how we calculate the common prefix, we'll have all of the data here - I don't think it saves us anything because we'd need to pull out those strings somewhere. Good point about the `get` usage - looking at it, I think we can nix it anyway since we can use ES5 getters - I'll look at that!